src/ert/run_models/base_run_model.py

```diff
@@ -613,6 +613,15 @@ def check_if_runpath_exists(self) -> bool:
         """
         return any(Path(run_path).exists() for run_path in self.paths)
 
+    def check_if_runpath_contains_data(self, iteration: int) -> bool:
+        target_folder = f"iter-{iteration}"
+
+        for run_path in self.paths:
```

Comment:
Think perhaps this is the root cause of the problems, `self.paths` should only return what is actually going to be run in, but suspect there is a bug there.